package/MDAnalysis/coordinates/chain.py

```diff
@@ -335,14 +371,12 @@ def _read_frame(self, frame):
     def _chained_iterator(self):
         """Iterator that presents itself as a chained trajectory."""
         self._rewind()  # must rewind all readers
-        readers = itertools.chain(*self.readers)
-        for frame, ts in enumerate(readers):
-            ts.frame = frame  # fake continuous frames, 0-based
-            self.ts = ts
-            # make sure that the active reader is in sync
-            i, f = self._get_local_frame(frame)  # uses 0-based frames!
-            self.__activate_reader(i)
-            yield ts
+        for i in range(self.n_frames):
+            j, f = self._get_local_frame(i)
+            self.__activate_reader(j)
+            self.ts = self.active_reader[f]
+            self.ts.frame = i
+            yield self.ts
```

Comment:
So this is needed to get correct iteration behavior for the continuous case. This is because I can't just go over all trajectories completely anymore but need to stop before I reach the end. I would like to know if I can make this possibly faster.

Reply:
Maybe have two code paths: one for the old behavior that uses `itertools.chain` and the other for continuous. At the moment we only allow continuous for trajectories that have random-access readers so we should be fine.